pkg/config/common.go

```diff
@@ -149,3 +149,14 @@ func ReadKey(keypath string) ([]byte, error) {
 
 	return data, nil
 }
+
+// NatsConfig is the configuration when using NATS as the event driver
+type NatsConfig struct {
+	// URL is the URL for the NATS server
+	URL string `mapstructure:"url" default:"nats://localhost:4222"`
+	// Prefix is the prefix for the NATS subjects to subscribe to
+	Prefix string `mapstructure:"prefix" default:"minder"`
+	// Queue is the name of the queue group to join when consuming messages
+	// queue groups allow multiple process to round-robin process messages.
+	Queue string `mapstructure:"queue" default:"minder"`
+}
```

Comment:
_If_ we want to move this up to `common` (rather than just having a split between `client` and `server` configuration), let's put it in its own file.  It feels kind of peculiar to have this in `common` while having the rest of the event configuration still in `server`, so I'd argue for moving the code back, particularly in the context of having smaller, more focused PRs.

Reply:
> If we want to move this up to common (rather than just having a split between client and server configuration), let's put it in its own file

That still breaks the pattern as all the structs in `common` don't have their own file. 

> so I'd argue for moving the code back,

To duplicate vs 'have common code'. Do you want me to duplicate the `NatsConfig` code in that case for `reminderconfig`

Reply:
I think there are a couple options here:

1. The way that structs are all piled into `common.go` today is probably a historical mistake.  I'm not asking you fix that mistake in this PR (because we should be making small, targeted changes), but we shouldn't make that mistake worse.  By putting this in `nats` or `eventing` files, it will make the future expansion more clear.
2. I think it's fine to have `pkg/config/reminder` depend on `pkg/config/server`, as long as we keep the graph acyclic.  In particular, if we have common server-side configuration for e.g. flags and messaging, it feels like it would make sense to have those be shared between the two.  Given that you can reasonably run a `minder` server without `reminder`, but the opposite is not very useful, I'd argue for having `pkg/config/reminder` import _parts_ of the config from `pkg/config/server`.  In the future, if we split the Minder server into e.g. `rpc_server` and `event_processor`, I'd suggest that both would want to use parts of the current `ServerConfig`, but would probably each have their own top-level struct for configuration (possibly in a separate module, or possibly in the `pkg/config/server` module).

I'm more concerned about leaking the dependencies on e.g. NATS into the `minder` client, which is statically linked and downloaded by many more users than the server.  Right now, this take that dependency, but I could see add a helper to create a NATS connection to this library in the future.